client-java/controller-api/src/main/java/org/evomaster/client/java/controller/api/dto/database/operations/MongoInsertionResultsDto.java

```diff
@@ -1,12 +1,39 @@
 package org.evomaster.client.java.controller.api.dto.database.operations;
 
 import java.util.ArrayList;
+import java.util.Collections;
 import java.util.List;
+import java.util.regex.Matcher;
+import java.util.regex.Pattern;
 
 public class MongoInsertionResultsDto {
     /**
      * whether the insertion at the index of a sequence of Mongo insertions (i.e., {@link MongoDatabaseCommandDto#insertions})
      * executed successfully
      */
     public List<Boolean> executionResults = new ArrayList<>();
+    public Integer failedInsertionIndex = -1;
+
+    public Boolean insertionFailed() {
+        return failedInsertionIndex != -1;
+    }
+
+    public void handleFailedInsertion(List<MongoInsertionDto> insertions, Exception e) {
+        failedInsertionIndex = findFailedInsertion(e);
+
+        if(insertionFailed()){
+            List<Boolean> results = new ArrayList<>(Collections.nCopies(insertions.size(), false));
+
+            for (int i = 0; i < failedInsertionIndex; i++) {
+                results.set(i, true);
+            }
+
+            executionResults = results;
+        }
+    }
+
+    private static int findFailedInsertion(Exception e) {
+        Matcher matcher = Pattern.compile("index (\\d+)").matcher(e.getMessage());
```

Comment:
explain this regex in a comment.
also, `Pattern.compile("index (\\d+)")` could be saved in a static variable, to avoid recompute it each time. but check if `matcher` is thread-safe

Reply:
Hi @arcuri82, in this [link](https://docs.oracle.com/javase/8/docs/api/java/util/regex/Pattern.html) says Instances of this class (referring to Pattern) are immutable and are safe for use by multiple concurrent threads. Instances of the Matcher class are not safe for such use. Does this mean we can store the Pattern in a static variable, but a new Matcher has to be created each time?